src/vehicle_gui_base.h

```diff
@@ -12,19 +12,53 @@
 #ifndef VEHICLE_GUI_BASE_H
 #define VEHICLE_GUI_BASE_H
 
+#include "core/smallvec_type.hpp"
+#include "date_type.h"
+#include "economy_type.h"
 #include "sortlist_type.h"
 #include "vehiclelist.h"
 #include "window_gui.h"
 #include "widgets/dropdown_type.h"
 
+#include <iterator>
+
 typedef GUIList<const Vehicle*> GUIVehicleList;
 
+struct GUIVehicleGroup {
+	VehicleList::const_iterator vehicles_begin;    ///< Pointer to beginning element of this vehicle group.
+	VehicleList::const_iterator vehicles_end;      ///< Pointer to past-the-end element of this vehicle group.
+	Money display_profit_this_year;                ///< Total profit for the vehicle group this year.
+	Money display_profit_last_year;                ///< Total profit for the vehicle group laste year.
+	Date age;                                      ///< Age in days of oldest vehicle in the group.
+	std::ptrdiff_t NumVehicles() const {
+		return std::distance(vehicles_begin, vehicles_end);
+	}
+	const Vehicle * GetSingleVehicle() const {
```

Comment:
more opening braces in the wrong place (there's quite a few others, i won't bother pointing them out)